avocado/plugins/assets.py

```diff
@@ -0,0 +1,179 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2019
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+import ast
+import errno
+import json
+import os
+import tempfile
+
+from avocado.utils import process
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+
+
+TEST_TEMPLATE = r"""
+from avocado import Test
+
+class FetchAssets(Test):
+    def test_fetch_assets(self):
+{fetch_assets}
+"""
+
+
+class FetchAssetHandler(ast.NodeVisitor):
+
+    def __init__(self, file_name):
+        self.file_name = file_name
+        # Right now, the class handles just 'fetch_assets' call.
+        # In the future, it can be extended to search for other call patterns.
+        self.pattern = 'fetch_asset'
+        self.calls = []
+        self.result = None
+
+        # create Abstract Syntax Tree from test source file
+        with open(self.file_name) as source_file:
+            self.tree = ast.parse(source_file.read(), self.file_name)
+
+        # build calls list
+        self.visit(self.tree)
+    # __init__()
+
+    def _build_call(self, node):
+        """
+        Build the call string from ast nodes.
+        :param node: Call root node
+        :type node: ast.Call
+        :return: Call prepared to be executed into a test class
+        :rtype: str
+        """
+        # we handle code indentation
+        call = "        self." + node.func.attr + "("
+        for a in node.args:
+            if isinstance(a, ast.Name):
```

Comment:
Still processing this loop, but I'd suggest:

1. using longer variable names (arg and kwarg?)
2. add comment before the `None` explaining that specifying arguments from variables is not yet supported. It's obvious now, but in 1, 2 or 10 years we'll have to go through the code and think about it again ;-)

PS: According to github statistics Czechs are the most chatty developers, we just like comments :-D

Reply:
> 1. using longer variable names (arg and kwarg?)
> 2. add comment before the `None` explaining that specifying arguments from variables is not yet supported. It's obvious now, but in 1, 2 or 10 years we'll have to go through the code and think about it again ;-)

Makes sense, will do.
 
> PS: According to github statistics Czechs are the chattiest developers, we just like comments :-D

And I really appreciate that! :D